### PR TITLE
Add subpixel smoothing example for metagrating problem

### PR DESCRIPTION
**Pull request by smartalecH**
Here, we use [fmmax](https://github.com/facebookresearch/fmmax), the [invrs.io gym](https://github.com/invrs-io/gym/tree/1c9a3f07ba78ea634bb9b6145596b3326dccb792), along with some handy [autograd-wrapper functions](https://github.com/mfschubert/agjax/tree/main) to take meep's recently added subpixel smoothing functionality, and optimize the metagrating example.

Importantly, this example shows how one can use fmmax (an RCWA/FMM tool) and still evolve β→∞.

We have an explicit shape optimization example:

image

As well as an explicit topology optimization example:

image

Everything was very straightforward thanks to all the existing code!

(cc @oskooi, @mfschubert, @stevengj, @mochen4)

### CODE REVIEW

**Comment by mfschubert** [comment]
Nice! FYI if you use `agjax.experimental.wrap_for_jax` ([code](https://github.com/mfschubert/agjax/blob/main/src/agjax/experimental/wrapper.py)) then you can also jit the entire calculation.

**Comment by smartalecH** [comment]
@mfschubert I wonder if there is a clean way to directly compute the vector fields from this formulation, which has a level set. In other words, we know the normal direction and distance to the nearest interface at every point, so perhaps this doesn't require solving a mini optimization problem?

**Comment by mfschubert** [comment]
@smartalecH This is an interesting thought, and in general a level set view may be fruitful to take. However, my sense is that we will still have to solve an optimization problem---recall that when computing the vector fields, we balance _smoothness_ with _alignment_ to the permittivity gradient. So, in some cases the optimal field is not aligned with the gradient, and a direct calculation from the permittivity (however it may be parameterized) may not produce such a field. But, maybe there is some cleverness that I am not thinking of this morning. Let's chat about it further.

**Comment by oskooi** [comment]
The optimal designs reported in the testbed paper (Section F and Fig. 8) have a diffraction efficiency of ~93%.

What are the values for the two designs shown above? What are the values of the designs computed using Meep?
